### DISCUSSION TITLE
How do I tell composer what version of Mediawiki I'm using?

### DISCUSSION THREAD

**Discussion by vapblack** [composer/composer#11997] [Q&A / Support]
Greetings. I'm overhauling my old mediawiki site, and I'm at the point where I am installing the extensions. When I use 
> composer update
I get the following error: Composer could not detect the root package (mediawiki/core) version, defaulting to '1.0.0'. See https://getcomposer.org/root-version

How can I make composer know what version of mediawiki I'm using?

**Comment by Seldaek** [comment]
You can set `"version": "..."` but you could also set the type to project or remove the name, that would suppress the warning. Because the warning really does not apply here.

**Reply by vapblack** [reply]
Yeah, I'm not sure where composer looks to get the version of the app its using.

**Reply by Seldaek** [reply]
Well if you click on the [link](https://getcomposer.org/doc/articles/troubleshooting.md#root-package-version-detection) in that warning message you posted, it is explained, but as I said in a case like mediawiki it is irrelevant as you should not have any circular dependencies.
